### ISSUE TITLE
Do not make additional calls for project/product templates on project page

### ISSUE THREAD

**Issue by maxceem** [appirio-tech/connect-app#2474]
After this [issue](https://github.com/appirio-tech/connect-app/issues/2472) is completed we can remove some additional calls for project and product templates on the project page and reuse already loaded data in the Redux Store.

Current Redux Store
```js
{
   projectState: {
      ...
      projectTemlate: {}, // current project's projectTemplate
      productTemplates: [],  // current project's productTemplates
      allProductTemplates: [], // all possbile productTemplates
   },
   templates: {
      projectTemplates: [...]   // all possible projectTemplates
      projectCategories: [...]  // all possible projectTypes
      isLoading: false
   }
}
```
What we can do:
1. Currently, project page loads one project template assosiated with it to `projectState.projectTemplate`making separate API call. As after previous issue is done we already have all `projectTemplates` in the store, we can find needed project template by its id. We have to make sure that templates are already loaded in the store at this point, and if not, then call action to load all templates.
   Note: after this we shouldn't have `projectState.projectTemplate` in the store anymore.

2. On the project page we load `projectState.allProductTemplates` with ALL product templates for `add phase` functionality. Also, we load  SOME product templates used in the project to  `projectState.productTemplates. Instead we should get product templates from `projectState.allProductTemplates` for a project using ids.
    Note: after this we shouldn't have `projectState.productTemplates` in the store anymore.

3. After previous two steps we still have `allProductTemplates` in `projectState` which is not a good place for it. As `projectState` suppose to keep some concrete project data, but `allProductTemplates` doesn’t depend on the concrete project. So it's better to move it to the`templates` section of the store and rename to `productTemplates`.
    Note: after this we shouldn't have `projectState.allProductTemplates` in store anymore, and have `template.productTemplates` instead.

After these steps Redux Store should look like this:
```js
{
   projectState: {
      ...
      // don't keep templates data anymore
   },
   templates: {
      // stay as before
      projectTemplates: [...]   // == /v4/projects/metadata -> projectTemplates
      projectCategories: [...]  // == /v4/projects/metadata -> projectTypes
      isLoading: false 
      // moved from projectState, before it was allProductTemplates
      productTemplates: [...] // == /v4/projects/metadata -> productTemplates
   }
}
```

fyi @RishiRajSahu

**Comment by arun1595** [comment]
@maxceem The first part of this issue #2472 has been fixed in #2494

I would like to fix this one once #2494 has been reviewed and merged.

**Comment by maxceem** [comment]
@arun1595 Yes, I'll try to review the other issue during today. So you could pick up this one. Thanks!

**Comment by maxceem** [comment]
@arun1595 I've merged your PR to the `cf12` branch so you can continue working with the changes you've already made.

**Comment by arun1595** [comment]
@maxceem Perfect, I'm on it.

**Comment by arun1595** [comment]
@maxceem I am a bit confused with the 2nd point.

>  Instead we should get product templates from projectState.allProductTemplates for a project using ids.

Shouldn't it be `templates.productTemplates` given that we've removed `projectState.allProductTemplates` in https://github.com/appirio-tech/connect-app/pull/2494?

**Comment by maxceem** [comment]
@arun1595 Yeah, you are right it should be `templates.productTemplates` as `projectState.allProductTemplates` was moved there in the previous issue.

**Comment by arun1595** [comment]
@maxceem Got it, thanks!

**Comment by maxceem** [comment]
@arun1595 so looks like you've already done point 3 in the previous issue PR #2494.
